enumeratum-core/src/main/scala/enumeratum/Enum.scala

```diff
@@ -38,7 +37,7 @@ trait Enum[A <: EnumEntry] {
   /**
     * Map of [[A]] object names to [[A]]s
     */
-  lazy final val namesToValuesMap: Map[String, A] =
+  lazy val namesToValuesMap: Map[String, A] =
```

Comment:
Nice. 

I think we should probably modify these `Map`s to be in terms of `namesToValuesMap` as well (I'd like to keep them final for now)

https://github.com/lloydmeta/enumeratum/blob/c895c327ba48e7646cf8c5815b505efd7d107fd6/enumeratum-core/src/main/scala/enumeratum/Enum.scala#L47-L54

Reply:
Can you clarify what you mean here?

Reply:
Right now, they're built in terms of `values` directly, whereas I think they should be built in terms of `namesToValuesMap` so that they can have the same values (in case the user overrides `namesToValuesMap` and wants to use functions that call `lowerCaseNamesToValuesMap`)

Reply:
I think I got it.